plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiOutputExtension.kt

```diff
@@ -155,14 +158,16 @@ class JvmAbiOutputExtension(
                                 if (descriptor != JvmAnnotationNames.METADATA_DESC)
                                     return delegate
                                 // Strip private declarations from the Kotlin Metadata annotation.
-                                return abiMetadataProcessor(delegate, removeCopyAlongWithConstructor)
+                                return abiMetadataProcessor(delegate, removeCopyAlongWithConstructor, doNotSortMembers)
                             }
 
                             override fun visitEnd() {
                                 // Output class members in sorted order so that changes in original ordering don't affect the ABI JAR.
 
-                                keptFields.sortedWith(compareBy(FieldNode::name, FieldNode::desc)).forEach { it.accept(cv) }
-                                keptMethods.sortedWith(compareBy(MethodNode::name, MethodNode::desc)).forEach { it.accept(cv) }
+                                if (!doNotSortMembers) {
+                                    keptFields.sortedWith(compareBy(FieldNode::name, FieldNode::desc)).forEach { it.accept(cv) }
```

Comment:
This probably disables more than then sorting? Doesn't this actually remove the fields and methods (since we never do the `it.accept(cv)` call) when we ask to not sort?

Reply:
PS: If that is right, I worry about the tests. Why does the test pass if members are indeed stripped here. Then it would seem that a different order would still lead to same ABI, but it seems like your test that they are different pass? 🤔

Reply:
😬  You are right, thank you for catching that, I'll take a look next week and see why tests didn't catch it. There is a chance that I just didn't run them >_<

Reply:
But it is weird =\ I wrote a test for a flag meaning that I ran tests at least twice to confirm the new test is working. It is easier to run all the tests than just a single one, so I suspect something is definetely wrong. I'll post what happened here.

Reply:
Alright, I guess I overreacted :) I thought that abi-gen tests didn't catch that I accidentally removed all methods/fields. That is not the case.

The problem with my tests for the flag: the test expects to get different api and at the first glance it looks like if we remove all methods/fields from two classes with the same name they should have the same abi instead. That is not the case because we still have `@Metadata` annotation which is still different.

Added content-based test, to catch it next time. Thank you for the review!